File: src/bin/target-sum.c
void helper(int* nums, int numsSize, int target, int* count) {
  if (numsSize == 0) {
    if (target == 0) {
      (*count)++;
    }
    return;
  }
  int n = *nums;
  helper(nums + 1, numsSize - 1, target - n, count);
  helper(nums + 1, numsSize - 1, target + n, count);
}

int findTargetSumWays(int* nums, int numsSize, int target) {
  int count = 0;
  helper(nums, numsSize, target, &count);
  return count;
}

File: src/bin/target-sum.rs
fn main() {
    println!("{}", Solution::find_target_sum_ways(vec![1, 1, 1, 1, 1], 3));
}

struct Solution;

impl Solution {
    pub fn find_target_sum_ways(nums: Vec<i32>, target: i32) -> i32 {
        fn helper(nums: &[i32], target: i32) -> i32 {
            if nums.len() == 0 {
                if target == 0 {
                    return 1;
                }
                return 0;
            }
            helper(&nums[1..], target + nums[0]) + helper(&nums[1..], target - nums[0])
        }

        helper(&nums[..], target)
    }
}
